### PR TITLE
Typescript SDK Integration

### LINKED ISSUE #43: feat: Integrate TypeScript SDK (spec 1.0) into diagram editor
### Description

Add the Serverless Workflow TypeScript SDK (spec 1.0) to the diagram editor package and implement a lightweight wrapper module as the single point of interaction with the SDK across the codebase. For now the wrapper should expose functions for ~deserialization~ and validation, along with a parse function that combines both into a single call. 

### Motivation

The diagram editor needs to accept and process Serverless Workflow definitions in YAML or JSON format in a spec-compliant way. Wrapping the SDK behind a thin abstraction decouples the codebase from SDK-specific implementation details, making it easier to adapt as the spec evolves. 

### Proposed Implementation

_No response_

### Definition of Done

- [x] Implementation: Fully implemented according to the Serverless Workflow spec.
- [x] Unit Tests: Comprehensive unit tests are included and passing.
- [ ] Integration Tests: Verified within the monorepo and target environments (Web/VS Code).
- [x] Documentation: Updated README.md, ADRs, or official docs.
- [ ] Performance: No significant regression in editor responsiveness.
- [ ] Accessibility: UI changes comply with accessibility standards.

### PR DESCRIPTION
**Pull request by lornakelly**
Closes: https://github.com/serverlessworkflow/editor/issues/43

## Summary                                                                                                                                                   
This PR integrates the `@serverlessworkflow/sdk` package into the diagram editor and creates an abstraction layer that isolates SDK dependencies and provides type-safe workflow parsing and validation.     
      
## Changes                                                                                                                                                                                                                                                     
- Added `@serverlessworkflow/sdk` and `js-yaml` dependencies to the project                                                                                              
- Created `workflowSdk.ts` abstraction layer with `validateWorkflow()` and `parseWorkflow()` functions                              
- Implemented type-safe error handling using `WorkflowParseResult`                                                                     
- Added integration tests to cover valid, invalid, and edge-case inputs in both YAML and JSON formats                                                                      
- Created test fixtures with valid/invalid workflows in YAML and JSON formats                                                                           
- Updated core module documentation explaining the abstraction layer's purpose        
-  Added `__snapshots__` to `.rat-excludes` to exclude generated snapshot files from license header checks

### CODE REVIEW

**Review by kumaradityaraj — APPROVED** [review]

**Review by fantonangeli — CHANGES_REQUESTED** [review]
Thanks a lot @lornakelly , I wrote a few comments regarding the tests

File: packages/serverless-workflow-diagram-editor/tests/core/workflowSdk.test.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import { deserializeWorkflow, validateWorkflow, parseWorkflow } from "../../src/core";
+import {
+  BASIC_VALID_WORKFLOW_YAML,
+  BASIC_VALID_WORKFLOW_JSON,
+  BASIC_INVALID_WORKFLOW_YAML,
+  BASIC_INVALID_WORKFLOW_JSON,
+} from "../fixtures/workflows";
+import { Classes, validate } from "@serverlessworkflow/sdk";
+
+vi.mock("@serverlessworkflow/sdk", () => ({
```
[the remark this answers is not in the document]
Comment (lornakelly) [reply packages/serverless-workflow-diagram-editor/tests/core/workflowSdk.test.ts]: Was thinking of this but ultimately decided that calling the sdk directly is more of an integration test than unit so stuck with simple unit tests for a start. I agree though, ill add a few tests that call the sdk directly as well

File: packages/serverless-workflow-diagram-editor/tests/core/workflowSdk.test.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  Classes: { Workflow: { deserialize: vi.fn() } },
+  validate: vi.fn(),
+}));
+
+const mockDeserialize = vi.mocked(Classes.Workflow.deserialize);
+const mockValidate = vi.mocked(validate);
+
+describe("deserializeWorkflow", () => {
```
[the remark this answers is not in the document]
Comment (lornakelly) [reply packages/serverless-workflow-diagram-editor/tests/core/workflowSdk.test.ts]: Fair point, since its such a thin wrapper, ill replace with some integration tests as well calling sdk directly

File: packages/serverless-workflow-diagram-editor/tests/core/workflowSdk.test.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  });
+
+  it("returns success:false and wraps non-Error thrown values in an Error", () => {
+    mockDeserialize.mockImplementation(() => {
+      throw "raw string";
+    });
+    const result = parseWorkflow(BASIC_INVALID_WORKFLOW_JSON);
+    expect(result.success).toBe(false);
+    if (result.success === false) {
+      expect(result.error).toBeInstanceOf(Error);
```
[the remark this answers is not in the document]
Comment (fantonangeli) [reply packages/serverless-workflow-diagram-editor/tests/core/workflowSdk.test.ts]: I can only think of this, but it only verifies `error` is an object containing a `message` with type string.
But at least we remove the `if`. Otherwise the 
```
    expect(result).toEqual({
      success: false,
      error: expect.objectContaining({
        message: "raw string"
      })
    });
```
or also here with the snapshot, the type is verified automatically:
```
    expect(result).toMatchSnapshot(); 
```

File: packages/serverless-workflow-diagram-editor/tests/core/workflowSdk.test.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+describe("parseWorkflow", () => {
+  beforeEach(() => vi.clearAllMocks());
+
+  it("returns success:true and a workflow for valid input", () => {
+    mockDeserialize.mockReturnValue(JSON.parse(BASIC_VALID_WORKFLOW_JSON));
+    mockValidate.mockReturnValue(undefined);
+    const result = parseWorkflow(BASIC_VALID_WORKFLOW_JSON);
+    expect(result.success).toBe(true);
+    if (result.success) {
```
[the remark this answers is not in the document]
Comment (lornakelly) [reply packages/serverless-workflow-diagram-editor/tests/core/workflowSdk.test.ts]: Used snapshots in integration tests, thanks for the suggestion!

**Review by handreyrc — APPROVED** [review]
@lornakelly,

It looks good to me.

Thanks for your work!

File: packages/serverless-workflow-diagram-editor/tests/core/workflowSdk.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (lornakelly) [reply packages/serverless-workflow-diagram-editor/tests/core/workflowSdk.test.ts]: Resolving as no longer applicable

**Review by fantonangeli — APPROVED** [review]
LGTM. Thank you @lornakelly .
I only left a minor comment, which can be skipped.

File: packages/serverless-workflow-diagram-editor/tests/core/workflowSdk.integration.test.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+import { Classes, Specification } from "@serverlessworkflow/sdk";
+
+describe("parseWorkflow", () => {
+  it.each([
+    { format: "YAML", input: BASIC_VALID_WORKFLOW_YAML, expectedName: "valid-workflow-yaml" },
+    { format: "JSON", input: BASIC_VALID_WORKFLOW_JSON, expectedName: "valid-workflow-json" },
+  ])("parses valid $format and returns model with no errors", ({ input, expectedName }) => {
+    const result = parseWorkflow(input);
+    expect(result.errors).toHaveLength(0);
+    expect(result.model?.document?.name).toBe(expectedName);
```
[the remark this answers is not in the document]
Comment (lornakelly) [reply packages/serverless-workflow-diagram-editor/tests/core/workflowSdk.integration.test.ts]: Yes I added this in as a safety check in case someone mistakenly updates the snapshot

**Review by ricardozanini — APPROVED** [review]
